littleblog/entry/serializers.py

```diff
@@ -1,15 +1,36 @@
 from rest_framework import serializers
 from .models import Entry, UserProfile
 from django.contrib.auth.models import User
+from rest_framework.validators import UniqueValidator
+from django.contrib.auth.password_validation import UserAttributeSimilarityValidator, MinimumLengthValidator, \
+    CommonPasswordValidator, NumericPasswordValidator
 
 
 class UserProfileCreateSerializer(serializers.ModelSerializer):
 
-    user = serializers.PrimaryKeyRelatedField(queryset=User.objects.all())
+    UserProfile = serializers.ReadOnlyField(source='profile.nickname')
```

Comment:
You sure you want to have a field with name "UserProfile" on your API? That's the name of db model, definitely not a name for API field.
You API should be consistent. Your fields are named in lowercase - all fields should be lowercase then.